scripts/client.py

```diff
@@ -47,9 +47,18 @@ def post(self, url, **kwargs):
     def put(self, url, **kwargs):
         kwargs_with_auth = self._add_auth_header(kwargs)
         r = requests.put(self.base_url + url, **kwargs_with_auth)
-        r.raise_for_status()
+        self.raise_for_status(r)
         return r.json()
 
+    @staticmethod
+    def raise_for_status(r: requests.Response):
+        if r.status_code > 399:
+            if r.headers["Content-Type"].startswith("application/json"):
```

Comment:
do we need to ignore case here?

Reply:
i meant the header key

Reply:
I believe the headers dict is case-insensitive but I'll double-check.

Reply:
Yeah, it's case-insensitive (changed it to `"CONTenT-TyPe"` locally and it still worked).